### PR TITLE
Board frontend

### PR DESCRIPTION
**Pull request by Robertito1**
### What this PR does (required):
- Used mock data to create the Kanban board
- Cards are draggable within a column and between them
- Columns are draggable

### Screenshots / Videos (required):
https://user-images.githubusercontent.com/60134828/127573279-1e58b923-99a4-4352-a733-4ac0aaf95468.mp4

### Any information needed to test this feature (required):
- Sign up or login
- Or Access the Dashboard as a demo user

### CODE REVIEW

File: client/src/components/Board/Board.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    setShowingWidgetRight(false);
+    const { destination, source } = result;
+    if (!destination) return;
+    if (destination.droppableId === source.droppableId && destination.index === source.index) {
+      return;
+    }
+    if (destination.droppableId === 'addColumnLeft' || destination.droppableId === 'addColumnRight') {
+      console.log('Adding');
+      return;
+    }
```
Comment (moffatethan) [inline client/src/components/Board/Board.tsx:29]: What is the purpose of this here? It's not doing anything but logging to the console.

**Review by moffatethan — APPROVED** [review]
Robert, this looks good. However, I think we could've create a context for the state of the board just in case we needed to access that information elsewhere in the application. Otherwise, please touch up the styles to match the mock-up and then this is good to go. 

Good work!

File: client/src/components/Board/Board.tsx (hunk 1 above)
Comment (Robertito1) [reply client/src/components/Board/Board.tsx:29]: That's correct @moffatethan this block does nothing here but log to the console, but I created this space to make it easy for anyone who is assigned to handle opening the dialog box for adding a new column (looks like me), currently on the branch I am working on, this `if` block does something and that is calling the function that opens up the dialog box for adding a new column to the board.

I probably should have used a comment line to make what I was doing a bit more apparent.

**Comment by Robertito1** [comment]
> Robert, this looks good. However, I think we could've create a context for the state of the board just in case we needed to access that information elsewhere in the application. Otherwise, please touch up the styles to match the mock-up and then this is good to go.

Thank you very much for your Feedback Ethan.

Utilizing Context API to persist the state is definitely in the books but I decided to hold off on that because the nature of the data from the backend might (very likely) change the logic of the way the state changes, causing double work, the state will be lost when the page is refreshed anyways so I thought that it could wait till we start integration

**Pull request merged**